packages/etcd/extra/etcd_discovery/etcd_discovery.py

```diff
@@ -334,9 +337,49 @@ def parse_cmdline() -> argparse.Namespace:
         default=detect_ip(),
         help="the IP address of the node to remove from the ensemble")
 
+    parser_ensure_perms = subparsers.add_parser('ensure-permissions')
+    parser_ensure_perms.set_defaults(func=ensure_permissions)
+
     return parser.parse_args()
 
 
+def ensure_permissions(args: argparse.Namespace) -> None:
+    log.info("ensure-permissions subcommand starts, args: `%s`", args)
+
+    etcdctl = EtcdctlHelper(
+        args.secure,
+        # NOTE(prozlach): we intentionally do not read back the nodes list from
+        # disk and connect to the local etcd instance. With more than one node
+        # in a quorum we would have sometimes intra-node and sometimes inter-node
+        # communication which may result in disruptions that happen only
+        # sometimes and are hard to reproduce.
+        [
+            "127.0.0.1",
+        ],
+        args.etcdctl_path,
+        args.ca_cert,
+        args.etcd_client_tls_cert,
+        args.etcd_client_tls_key,
+    )
+
+    # See below for more context:
+    # https://github.com/etcd-io/etcd/blob/3898452b5432b4d69028ee79d796ddeab0acc42c/Documentation/op-guide/authentication.md
+    etcdctl.ensure_user("root")
+    etcdctl.grant_role("root", "root")
+
+    etcdctl.ensure_user("calico")
```

Comment:
I have a feeling that the list of initial users and roles must be placed in a separate file that would be easy for someone to find out and update. But I understand the equally important role of not over-engineering stuff, that leads to accepting the fact that someone else could simply update this file with the new users/roles if needed 😄